target/linux/lantiq/dts/RS353.dts

```diff
@@ -0,0 +1,318 @@
+/dts-v1/;
+
+/include/ "vr9.dtsi"
+
+/ {
+	model = "RS353 - Bintec Elmeg RS353";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200 init=/etc/preinit";
+		leds {
+                        running = &status;
+                        boot = &status;
+			wifi = &wlan;
+			usb2 = &usb; /* Front port is mapped to device 2 */
+			dsl = &dsl;
+                };
+
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	sram@1F000000 {
+		cgu0: cgu@103000 {
+                        compatible = "lantiq,cgu-xway";
+                        reg = <0x103000 0x1000>;
+                        #clock-cells = <1>;
+			lantiq,phy-clk-src = <0x2>;
+		};
+	};
+
+	fpi@10000000 {
+		localbus@0 {
+			nor-boot@0 {
+				compatible = "lantiq,nor";
+				bank-width = <2>;
+				reg = <0 0x0 0x2000000>;
+				#address-cells = <1>;
+				#size-cells = <1>;
+
+				partitions {
+					compatible = "fixed-partitions";
+					#address-cells = <1>;
+					#size-cells = <1>;
+
+					boardconfig: partition@0 {
```

Comment:
I think the original firmware runs in little endian and the original u-boot  in big endian. But, the RAM content seen by the original firmware is identical to the ram content seen by u-boot or Linux kernel. Only the flash content is reverser between each.

In my opinion, the kernel relocation is the better option. I used U-boot for simplicity and to be sure I can launch it from the original Bootmonitor.

For testing purpose, I can run directly an update image from RAM. When you transfer an update image by tftp to the Bootmonitor, It launchs it directly from RAM (as the update image contains a mini-flasher). In this case, the image is started from 0x05FB0034

The image length in the header determines the length for crc32 calculation (as you said), but it also determines the length of image to be really writed to the flash. So, I can't use it to bypass the crc32 calculation by this way. Also, I can patch and reflash the bootmonitor to boot even if the crc32 is not correct.